src/Widgets/MaskedImage.vala

```diff
@@ -0,0 +1,70 @@
+/*
+* Copyright 2019 elementary, Inc. (https://elementary.io)
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public
+* License as published by the Free Software Foundation; either
+* version 3 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have received a copy of the GNU General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1301 USA
+*
+*/
+
+public class Notifications.MaskedImage : Gtk.Overlay {
+    private const int ICON_SIZE = 48;
+    public string image_path { get; construct; }
+
+    public MaskedImage (string image_path) {
+        Object (image_path: image_path);
+    }
+
+    construct {
+        var mask = new Gtk.Image.from_resource ("/io/elementary/notifications/image-mask.svg");
+        mask.pixel_size = ICON_SIZE;
+
+        var image = new Gtk.Image ();
+        image.pixel_size = ICON_SIZE;
+
+        try {
+            var scale = get_style_context ().get_scale ();
+            var pbuf = new Gdk.Pixbuf.from_file_at_size (image_path, ICON_SIZE * scale, ICON_SIZE * scale);
+
+            image.gicon = mask_pixbuf (pbuf, scale);
+        } catch (Error e) {
+            critical ("Unable to mask image: %s",e.message);
```

Comment:
If we fail to load the image_path for whatever reason (path doesn't exist etc.), perhaps we should fallback to the default icon / app icon?

Reply:
Yeah I agree. I'm not certain how to do that though. Should we self destruct this widget and have a null check in Bubble.vala?